README.md
![WOWEN Tech Logo](https://github.com/WOWEN-DEV/wowen.tech/blob/main/assets/img/wowen-tech-repo-logo.svg)

<p align="center">
  <a href="https://app.netlify.com/sites/wowen/deploys">
    <img src="https://api.netlify.com/api/v1/badges/b86ae6b4-7487-4cc1-b96a-0925766438c0/deploy-status" alt="Netlify Status">
  </a>
  <a href="https://github.com/WOWEN-DEV">
    <img src="https://img.shields.io/badge/WOWEN-FemTech%20Dev%20Community-blueviolet" alt="WOWEN">
  </a>
</p>

<h1 align="center">WOWEN - Transforming FemTech</h1>

Welcome to the WOWEN website repository! We're thrilled to have you here as we embark on a journey to revolutionize FemTech. 🚀

<br>
<br>
<div align="center">
  <img src="https://www.wowen.tech/assets/img/hands.svg" alt="WOWEN Illustration">
</div>


# Table of Contents
- [About](#about)
- [Deployment and Staging](#deployment-and-staging)
  - [Never deploy and commit directly to main branch](#never-deploy-and-commit-directly-to-main-branch)
  - [Branches and deploy contexts](#branches-and-deploy-contexts)
  - [Current settings and deploy contexts in Netlify](#current-settings-and-deploy-contexts-in-netlify)
- [Changelog](#changelog)
- [Roadmap](#roadmap)
- [Developer Documentation: Technical Architecture and Stack Overview](#developer-documentation-technical-architecture-and-stack-overview)
  - [Frontend](#frontend)
  - [Domain Management](#domain-management)
  - [Hosting and Deployment](#hosting-and-deployment)
  - [Form Collection](#form-collection)
  - [Analytics](#analytics)
  - [Future Considerations](#future-considerations)
  - [Summary of Technologies](#summary-of-technologies)
- [Contributing](#contributing)
- [Feedback and Support](#feedback-and-support)
- [Connect with Us](#connect-with-us)

<br>

# About
WOWEN is not just a company; we are a creative powerhouse with expertise in full-stack development, FemTech, UX, design, and marketing. We have a unique ability to transform high-level strategy into practical implementation at the intersection of business development, communication, and technology.

<br>

# Deployment and Staging

Our website is hosted and deployed with Netlify. Production site [www.wowen.tech](https://www.wowen.tech/) is `main` branch, staging site [staging--wowen.netlify.app](https://staging--wowen.netlify.app) is `staging` branch.

## Never deploy and commit directly to main branch
In most cases, changes should never be directly deployed to our `main` branch, which serves as the production environment/main branch/site. Instead, commit and deploy your changes to our staging branch first, and if approved it will be merged and synced into our `main` branch. For the website ([www.wowen.tech](https://www.wowen.tech/)), the staging branch is named `staging` and can be accessed at [https://staging---wowen.netlify.app](https://staging--wowen.netlify.app).

## Branches and deploy contexts

For those who are new with the terms:
- **Branch:** A branch is like a version of our code, our `main` branch is our Web App/production site [https://www.wowen.tech](https://www.wowen.tech) and our `staging` branch is our staging site [https://staging--wowen.netlify.app](https://staging--wowen.netlify.app) where we stage/test updates before they end up on our production site, the `main` branch. This is a way to work on updates or new features without affecting the publicly known and used live website. Once you're satisfied with the updates on `staging`, you can combine ("merge") the `staging` branch into the `main` branch, the main version of the code, our production site. 
- **Staging:** Is a separate test version of our site. Our branch `staging` is our staging enviroment. Current branch update/status/deployment is always accessed live at [https://staging--wowen.netlify.app](https://staging--wowen.netlify.app).
- **Pull Request:** Is a proposal to update some part of our website's code, if the proposal is good, it will be "merged" into `main` branch, our production site [https://www.wowen.tech](https://www.wowen.tech).
- **Merging:** Merging is the process of taking the changes from one branch (version) of our code and combining them with another branch, usually the `main` one. This allows us to update the official version of our project with new features, fixes, or improvements that we've been working on separately on staging. Once merged, those changes become part of the `main` or target branch and are reflected on [https://www.wowen.tech](https://www.wowen.tech).
- **Deploy Contexts:** Are settings that tell Netlify how to handle our different branches when turning them into live websites.
- **Deploy Previews:** Are temporary websites that Netlify creates to show you what these changes will look like if they go live. When someone suggests changes to our code, they often make a "pull request." With this setting, Netlify will automatically create a temporary, live version of our site incorporating those suggested changes. This is called a "Deploy Preview." This allows you to preview changes before they go live on our main site (`main` branch).
- **Branch Deploys:** The `main` branch serves as our production branch, while branch deploys encompass all other branches created here that we designate to be included as "Branch Deploys."

## Current settings and deploy contexts in Netlify
- **Production Branch:** `main`
- **Branch Deploys:** `staging` *Note: Netlify is with this setting ignoring any other branches we create in this repo; it won't make websites from them*.
- **Deploy Previews:** `Any pull request against your production branch / branch deploy branches`

<br>

# Changelog
This changelog provide a high-level overview of the changes made in each version of the website:
- [Changelog](https://github.com/WOWEN-DEV/wowen.tech/blob/main/CHANGELOG.md)

<br>

# Roadmap

At present, we do not have a roadmap for wowen.tech website.

<br>

# Developer Documentation: Technical Architecture and Stack Overview
Our technical infrastructure is designed to be robust and scalable, leveraging a variety of technologies and services. Below is a detailed breakdown of our current setup:

## Frontend

- **Technologies**: HTML, CSS, JavaScript
- **Description**: The frontend is a simple setup consisting of HTML, CSS, and JavaScript files.

## Domain Management

- **Registrar**: One.com
- **Description**: We've registered our domain (`wowen.tech`) and all associated subdomains with One.com. As of now, we intend to continue using One.com as our domain registrar due to its other convenient features.

## Hosting and Deployment

- **Platform**: Netlify
- **Description**: We use Netlify for hosting and deployment. We've set up the DNS settings on One.com to point to Netlify, which serves as the platform for both our frontend hosting and deployment.

## Form Collection

- **Netlify Forms**: Used to collect contact forms on `/contact` and `/sv/kontakt`.
- **Tally**: Used to collect applicant forms on `/careers` and `/sv/jobb/`.

## Analytics

- **Google Analytics**: Used for web traffic analysis – (`WOWEN – wowen.tech – GA4`).
- **Plausible**: An additional tool for web traffic analysis – (`WOWEN – wowen.tech`).
- **Google Tag Manager**: Used for managing pixels and tags on the website, current implemented tags: *Google Analytics*, *Facebook Pixel*, *LinkedIn Insight Tag* – (`WOWEN – www.wowen.tech`)

## Future Considerations

As of now, we plan to continue using Netlify for hosting and deployment and One.com for domain management. However, we are evaluating other options for hosting and domain management, potentially transitioning to AWS Services or Vercel.

## Summary of Technologies
- One.com (Domain management + database possibilities + archive for old email domain wowen.se)
- Netlify (Hosting, frontend deployment, form collection, authentication)
- Tally (Form builder)

<br>

# Contributing

We welcome contributions from the FemTech Dev Community. If you'd like to contribute to our projects, please follow our [Contributing Guidelines](CONTRIBUTING.md).

<br>

# Feedback and Support

Have feedback or need support? Feel free to reach out!

<br>

# Connect with Us

- **Website:** [wowen.tech](https://wowen.tech)
- **First Contact Email:** emma@wowen.tech

<br>

---

Thank you for your interest in WOWEN. Together, we're making FemTech innovative, creative, and powering. 💪 

Copyright © 2019-2023, WOWEN. All rights reserved.


CHANGELOG.md
# Changelog

All notable changes to this website will be documented in this file.

## v1.0.0 – YYYY-MM-DD

### Added
- None.

### Changed
- None.

### Deprecated
- None.

### Removed
- None.

### Fixed
- None.
  
### Security
- None.




legacy/lifertility/assets/js/menu﹖ver=6.2.2.js
/* CLICK MENU */

$('.menu-open').on('click', function(){

	$('.overlay.scroll').addClass('open');
	$('.menu-content').addClass('open');
	$('.c-scrollbar').addClass('open');
	$('.menu-content .follow').addClass('open');
	$('.menu-content .coordonnees').addClass('open');
	$('.menu-content .menu-links').addClass('open');
	$('.menu-open').addClass('disable');
	$('.menu-close').addClass('disable');
	
});

$('.menu-close').on('click', function(){
	
	$('.overlay.scroll').removeClass('open');
	$('.menu-content').removeClass('open');
	$('.c-scrollbar').removeClass('open');
	$('.menu-content .follow').removeClass('open');
	$('.menu-content .coordonnees').removeClass('open');
	$('.menu-content .menu-links').removeClass('open');
	$('.menu-open').removeClass('disable');
	$('.menu-close').removeClass('disable');
	
});

legacy/lifertility/assets/js/script﹖ver=6.2.2.js

/* PRÊT */

$(document).ready(function(){
	
	$(this).scrollTop(0);
	setTimeout(function(){$('html').addClass('is-ready');}, 1000);
	$('main').addClass('overflow');
	setTimeout(function(){$('#loader').addClass('loading');}, 300);
	setTimeout(function(){$('main').removeClass('overflow');}, 1000);
	
});

$(window).on('beforeunload', function(){

	$('body').addClass('unload');

});

$('.overlay-menu--cta').hover(function () {$('.container').addClass('menu');}, function () {$('.container').removeClass('menu');});



/* SCROLLER */
	
if (window.matchMedia("(min-width: 790px)").matches) {
	
	setTimeout(() => {
		
		const map = (x, a, b, c, d) => clamp((x - a) * (d - c) / (b - a) + c, Math.min(c,d), Math.max(c,d));
		const clamp = (num, min, max) => num <= min ? min : num >= max ? max : num;
		const images = [...document.querySelectorAll('.scale')];
			
		const scroller = new LocomotiveScroll({
			el: document.querySelector('[data-scroll-container]'),
			smooth: true,
			inertia:.75,
			smartphone: {smooth: true},
			tablet: {smooth: true}
		});
		
		scroller.on('scroll', (position) => {
			
		  if ((position.scroll.y) > 200) {
			  
			document.querySelector('.overlay').classList.add('scroll');
			
		  } else {
			  
			document.querySelector('.overlay').classList.remove('scroll');
		  
		  }
		  
		});

		scroller.on('scroll', (position) => {
			
		  if ((position.scroll.y) > 0) {
			  
			document.querySelector('.rounded svg').style.transform = 'rotate(' + position.scroll.y/3 + 'deg)';
			
		  } else {
			  
			document.querySelector('.rounded svg').style.transform = 'rotate(0deg)';
			
		  }
			
		});
		
		document.querySelector('.footer-top').addEventListener('click', (function(){scroller.scrollTo(0)}));

		scroller.update();
		
	}, 1000)
	
} else {
	
	$(window).scroll(function() {    
    var scroll = $(window).scrollTop();

		if (scroll > 80) {
			$(".overlay").addClass("scroll");
			
		} else {
			  
			$(".overlay").removeClass("scroll");
		  
		 }
	});
	
}



/* CAROUSSEL */

const caroussel = new Swiper('.caroussel', {
	
	direction: 'horizontal',
	slidesPerView: 1,
	speed:700,
	spaceBetween:50,
	
	pagination: {
		el: '.galerie-pagination',
		type: 'fraction',
	},
	
	navigation: {
	nextEl: '.nav-right',
	prevEl: '.nav-left',
	},
	
	breakpoints: {
		790: {slidesPerView: 2,},
		1290: {slidesPerView: 1.3,},
	}

});

const realisations = new Swiper('.realisations-caroussel', {
	
	direction: 'horizontal',
	slidesPerView: 1,
	speed:500,
	spaceBetween:50,
	
	pagination: {
	  el: ".navigation-pagination",
	  type: "progressbar",
	},
	
	breakpoints: {
		790: {slidesPerView: 2, spaceBetween: 50, speed:1000,},
		1090: {slidesPerView: 3, spaceBetween: 100, speed:1000,},
	}

});







legacy/fertilai/src/js/main.js
Prism.highlightAll();

splt({
  reveal: true,
});

const navBackground = document.querySelector('header');

window.onload = () => {
  const heroText = document.querySelector('.hero-text');
  heroText.style.opacity = '1';
  navBackground.style.opacity = '1';
  let heroTextAnimation = anime({
    targets: '.hero-text.splt .reveal',
    translateY: [64, 0],
    duration: 700,
    delay: anime.stagger(15, { start: 250 }),
    easing: 'cubicBezier(.6,-1.5,.1,1.9)',
  });

  let npmAnimation = anime({
    targets: '#copy-npm',
    translateY: [8, 0],
    scale: [0.9, 1],
    opacity: [0, 1],
    duration: 800,
    delay: 900,
    easing: 'cubicBezier(.6,-1.5,.1,1.9)',
  });

  let valueCardAnimations = anime({
    targets: '.value',
    translateY: [24, 0],
    opacity: [0, 1],
    scale: [0.9, 1],
    duration: 900,
    delay: anime.stagger(200, { start: 500 }),
    easing: 'cubicBezier(.6,-1.5,.1,1.9)',
  });
};

// logo animation and trigger + throttling
splt({
  target: '.version',
  reveal: true,
});

const logo = document.getElementById('logo');
logo.addEventListener('mouseenter', () => {
  if (anime.running.length > 1) {
  } else {
    let versionAnimation = anime({
      targets: '.version .reveal',
      translateY: [
        { value: [0, -32], duration: 400, delay: anime.stagger(15, { start: 150 }) },
        { value: [32, 0], duration: 600, delay: anime.stagger(15) },
      ],
      easing: 'cubicBezier(.6,-1.5,.1,1.5)',
    });
  }
});

// npm click animation card creation setup
let copiedMsgCreate = () => {
  let messageContainer = document.createElement('div');
  messageContainer.classList.add('message-container');
  let copiedMessage = document.createElement('div');
  copiedMessage.classList.add('copied-message');
  copiedMessage.innerHTML = 'Kopierat till urklipp!';
  messageContainer.appendChild(copiedMessage);
  let copiedArrow = document.createElement('div');
  copiedArrow.classList.add('copied-arrow');
  messageContainer.appendChild(copiedArrow);
  copyNPM.appendChild(messageContainer);
  setTimeout(() => {
    messageContainer.style.opacity = 1;
  }, 1);
  setTimeout(() => {
    messageContainer.style.opacity = 0;
    setTimeout(() => {
      messageContainer.remove();
    }, 200);
  }, 1200);
};

// copies npm text to clipboard
const copyNPM = document.getElementById('copy-npm');
const npmIcon = document.querySelector('.npm-icon');

// click function for NPM animation
copyNPM.addEventListener('click', (e) => {
  e.preventDefault;
  navigator.clipboard.writeText('git clone https://github.com/WOWEN-DEV/fertilai.git');
  copiedMsgCreate();

  npmIcon.classList.remove('is_animating');

  void npmIcon.offsetWidth;
  npmIcon.classList.add('is_animating');
});

// features section animations

let navColor = 'var(--primary-3)'; // determines navigation color shift end result
let navIsIntersecting = false; // stops nav from changing colors if not intersecting

splt({
  target: '.demo',
  reveal: true,
});
let noSPLT = false; // primer for revert in animation 3

let animation;
let exampleAnimation = (x) => {
  if (x == 0) {
    animation = anime({
      targets: '.demo .char',
      translateY: [0, -30],
      duration: 300,
      loop: true,
      direction: 'alternate',
      delay: anime.stagger(30, { start: 500 }),
      easing: 'cubicBezier(.64,-0.38,.43,1.54)',
    });
    noSPLT = false;
  } else if (x == 1) {
    animation = anime({
      targets: '.demo .reveal',
      translateY: [72, 0],
      duration: 600,
      loop: true,
      direction: 'alternate',
      delay: anime.stagger(20, { start: 500 }),
      endDelay: 300,
      easing: 'cubicBezier(.64,-0.38,.43,1.54)',
    });
    noSPLT = false;
  } else if (x == 2) {
    animation = anime({
      targets: '.demo .reveal',
      translateY: [72, 0],
      duration: 600,
      delay: anime.stagger(10, { start: 500 }),
      easing: 'cubicBezier(.64,-0.38,.43,1.54)',
      complete: () => {
        anime({
          targets: '.demo #c1, .demo #c2, .demo #c3',
          keyframes: [{ translateY: -80 }, { scale: 0.8 }, { rotate: 35 }, { rotate: 0 }, { scale: 1 }, { translateY: 0 }],
          duration: 3800,
          endDelay: 100,
          delay: anime.stagger(80, { start: 500 }),
          easing: 'cubicBezier(1,-1.6,.35,2.06)',
          complete: () => {
            anime({
              targets: '.demo .reveal',
              translateY: [0, -72],
              duration: 600,
              delay: anime.stagger(10, { start: 500 }),
              easing: 'cubicBezier(.64,-0.38,.43,1.54)',
              complete: animation.restart,
            });
          },
        });
      },
    });
    noSPLT = false;
  } else if (x == 3) {
    animation = anime({
      targets: '.demo .char',
      translateY: () => {
        return anime.random(-30, 30);
      },
      duration: 500,
      direction: 'alternate',
      loop: 1,
      delay: anime.stagger(50, { from: 'center', start: 500 }),
      easing: 'cubicBezier(.71,-5,.43,5)',
      complete: () => {
        splt.revert();
        noSPLT = true; // primer triggered, next click will resplit
      },
    });
  }
};
exampleAnimation(0);

//

// reset reveal position
let revealReset = () => {
  const reveals = document.querySelectorAll('.reveal');
  for (let i = 0; i < reveals.length; i++) {
    reveals[i].style.transform = 'translateY(0px)';
  }
};

// click functions for features section
const featuresCard = document.querySelectorAll('.feature');
for (let i = 0; i < featuresCard.length; i++) {
  featuresCard[i].addEventListener('click', () => {
    // resplit characters
    if (noSPLT == true) {
      splt({
        target: '.demo',
        reveal: true,
      });
    }
    // // reset active modifiers
    for (let e = 0; e < featuresCard.length; e++) {
      featuresCard[e].classList.remove('active');
    }

    // resets animations to 0 to cleanly play them with each click
    if (anime.running.length > 0) {
      animation.restart();
    }
    anime.running.length = 0;

    // specific color selections
    if ([i] == 0) {
      revealReset();
      exampleAnimation(0);
    } else if ([i] == 1) {
      exampleAnimation(1);
    } else if ([i] == 2) {
      exampleAnimation(2);
    } else if ([i] == 3) {
      revealReset();
      exampleAnimation(3);
    }
    // set new active modifiers
    featuresCard[i].classList.add('active');
  });
}

const observer = new IntersectionObserver(
  (entries) => {
    // Loop over the entries
    entries.forEach((entry) => {
      // If the element hits trigger
      if (entry.isIntersecting) {
        //console.log('is intersecting');
        navIsIntersecting = true;
        document.querySelector('header').style.background = navColor;
      } else {
        //console.log('is not intersecting');
        navIsIntersecting = false;
        document.querySelector('header').style.background = 'var(--gray-4)';
      }
    });
  },
  { rootMargin: '-10% 0px -90% 0px' }
);

observer.observe(document.querySelector('.features'));

// brgr icon menu
let brgrIconClicked = false;
const brgrIconWrapper = document.querySelector('.brgr-icon-wrapper');
const topLine = document.querySelector('.brgr-icon-line-1');
const bottomLine = document.querySelector('.brgr-icon-line-2');
const menu = document.querySelector('.menu-s');
const body = document.querySelector('body');

let menuAnimation = () => {
  anime({
    targets: '.menu-link-s',
    translateY: [24, 0],
    opacity: [0, 1],
    duration: 500,
    delay: anime.stagger(50),
    easing: 'cubicBezier(.6,-1.5,.1,1.9)',
  });
};

let brgrIconClickTrue = () => {
  topLine.style.top = '50%';
  topLine.style.transform = 'rotate(45deg)';
  bottomLine.style.top = '50%';
  bottomLine.style.transform = 'rotate(-45deg)';
  navBackground.style.background = 'var(--white)';
  body.style.height = '100vh';
  body.style.overflow = 'hidden';
  menu.style.display = 'flex';
  setTimeout(() => {
    menu.style.opacity = '1';
  }, 0);
  logo.style.color = 'var(--white)';
  menuAnimation();
  brgrIconClicked = true;
};

let brgrIconClickFalse = () => {
  topLine.style.top = '35%';
  topLine.style.transform = 'rotate(0deg)';
  bottomLine.style.top = '65%';
  bottomLine.style.transform = 'rotate(0deg)';
  if (navIsIntersecting == true) {
    navBackground.style.background = 'var(--primary-3)';
  } else {
    navBackground.style.background = 'var(--gray-4)';
  }
  body.style.height = 'auto';
  body.style.overflow = 'visible';
  menu.style.opacity = '0';
  setTimeout(() => {
    menu.style.display = 'none';
  }, 300);
  brgrIconClicked = false;
};

brgrIconWrapper.addEventListener('click', () => {
  if (brgrIconClicked == false) {
    brgrIconClickTrue();
  } else {
    brgrIconClickFalse();
  }
});

const menuItems = document.querySelectorAll('.menu-link-s');

for (let i = 0; i < menuItems.length; i++) {
  menuItems[i].addEventListener('click', () => {
    brgrIconClickFalse();
  });
}

logo.addEventListener('click', () => {
  brgrIconClickFalse();
});

// added to kick the user out of brgr icon mode should the browser be resized
window.addEventListener('resize', () => {
  brgrIconClickFalse();
});

// support the creator button functionality

const closeBtn = document.querySelector('.x');
const supportCard = document.querySelector('.support-card');

setTimeout(() => {
  supportCard.style.display = 'flex';
  let supportIn = anime({
    targets: '.support-card',
    translateY: [32, 0],
    translateX: ['-50%', '-50%'],
    rotate: [8, 0],
    opacity: [0, 1],
    duration: 900,
    easing: 'cubicBezier(.6,-1.5,.1,1.9)',
  });
}, 10000);

closeBtn.addEventListener('click', () => {
  let supportOut = anime({
    targets: '.support-card',
    translateY: [0, 32],
    translateX: ['-50%', '-50%'],
    rotate: [0, 4],
    opacity: [1, 0],
    duration: 900,
    easing: 'cubicBezier(.6,-1.5,.1,1.9)',
    complete: () => {
      supportCard.style.display = 'none';
    },
  });
});

// guides demo functionality

let running = false; // throttles button clicks

const demoOne = document.getElementById('demo-1');
demoOne.addEventListener('click', () => {
  if (running == false) {
    running = true;
    anime({
      begin: splt({
        target: '.demoOneSplt',
      }),
      targets: '.demoOneSplt .char',
      translateY: [0, -20],
      direction: 'alternate',
      loop: 1,
      delay: anime.stagger(25),
      easing: 'cubicBezier(.71,-0.77,.43,1.67)',
      complete: () => {
        splt.revert();
        running = false;
      },
    });
  } else {
  }
});

const demoTwo = document.getElementById('demo-2');
demoTwo.addEventListener('click', () => {
  if (running == false) {
    running = true;
    anime({
      begin: splt({
        target: '.demoTwoSplt',
        reveal: true,
      }),
      targets: '.demoTwoSplt .reveal',
      translateY: [0, 20],
      direction: 'alternate',
      loop: 1,
      delay: anime.stagger(25),
      easing: 'cubicBezier(.71,-0.77,.43,1.67)',
      complete: () => {
        splt.revert();
        running = false;
      },
    });
  } else {
  }
});

const demoThree = document.getElementById('demo-3');
demoThree.addEventListener('click', () => {
  if (running == false) {
    running = true;
    anime({
      begin: splt({
        target: '.demoThreeSplt',
        reveal: true,
      }),
      targets: '.demoThreeSplt #c1, .demoThreeSplt #c2 #r',
      translateY: [0, -20],
      direction: 'alternate',
      loop: 1,
      delay: anime.stagger(50),
      easing: 'cubicBezier(.71,-0.77,.43,1.67)',
      complete: () => {
        splt.revert();
        running = false;
      },
    });
  } else {
  }
});

const demoFour = document.getElementById('demo-4');
demoFour.addEventListener('click', () => {
  if (running == false) {
    running = true;
    anime({
      begin: splt({
        target: '.demoFourSplt',
      }),
      targets: '.demoFourSplt .char',
      opacity: [1, 0],
      scale: [1, 1.5],
      direction: 'alternate',
      loop: 1,
      delay: anime.stagger(50),
      easing: 'cubicBezier(.71,-0.77,.43,1.67)',
      complete: () => {
        splt.revert();
        running = false;
      },
    });
  } else {
  }
});
